tests/test_sanity.py

```diff
@@ -241,3 +243,42 @@ def test_from_container_spawned_process(
         assert_collapsed(collapsed)
 
     assert b"Profiling spawned process" in container.logs()
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("profiler_type", ["ap"])
+@pytest.mark.parametrize("runtime", ["java"])
+@pytest.mark.parametrize("application_image_tag", ["musl"])
+def test_container_name_when_stopped(
+    docker_client: DockerClient,
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    runtime_specific_args: List[str],
+    profiler_type: str,
+    profiler_flags: List[str],
+    application_docker_container: Container,
+) -> None:
+    """
+    Tests that container name is added to data even when container stops during profiling.
+    Related issue: https://github.com/Granulate/gprofiler/issues/640
+    """
+    profiler_flags.extend(["-d", "20"])
+    container = start_gprofiler_in_container_for_one_session(
+        docker_client, gprofiler_docker_image, output_directory, output_collapsed, runtime_specific_args, profiler_flags
+    )
+    try:
+        wait_event(
+            20,
+            Event(),
+            lambda: re.search(rb"Profiling .*process \d* with async-profiler", container.logs()) is not None,
```

Comment:
I know it was my suggestion but I realized it makes no sense to use a Java app here - once the container is dead, async-profiler dies too, as it lives in the Java app... so how can we read its output (it writes the output in the container, but it's dead)? lol. Did you get this test to pass? Because I think it should not.

Another suggestion - let's use a Python app with `profiler_type=py-spy` - it prints an indicative message upon process exit, that is dumped into the profiler log:
```
py-spy> Stopped sampling because process exited
```

You can kill the app, then search for this message.

As a side note, to ensure this test covers exactly the case we want to test, you can build gProfiler from master and run it - it should fail on master and pass on your branch.

Reply:
Sure I will change it to py-spy. I was able to pass both java and nodeJs test on this PR and I tested both on master and both failed :)

Reply:
Changed, Also works on this PR fails on master

Reply:
Awesome